src/Dashboard/ApiControllers/FunctionsController.cs

```diff
@@ -626,6 +627,12 @@ public IHttpActionResult Abort(string instanceQueueName)
 
         private bool? HostInstanceHasHeartbeat(FunctionInstanceSnapshot snapshot)
         {
+            if (snapshot.HeartbeatExpiredTime.HasValue)
+            {
+                var now = DateTime.UtcNow;
+                return snapshot.HeartbeatExpiredTime.Value > now;
```

Comment:
Don't we want the other way around? I.e. on Refresh we add 50 minutes to UtcNow, so we're expired only when now catches up to that value?